packages/integration-aws-node/src/config.ts

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+
+/**
+ * The configuration parameters for a single AWS account for the AWS integration.
+ *
+ * @public
+ */
+export type AwsIntegrationAccountConfig = {
```

Comment:
Could this also be sourced via the catalog, analog to https://backstage.io/docs/features/kubernetes/configuration#catalog / https://github.com/backstage/backstage/issues/13727 ?

Having AWS Accounts in the Catalog as Resource Entities might be useful in the future.

Reply:
There has been some discussion in the RFC (https://github.com/backstage/backstage/issues/12844) about having an AWS account discovery plugin (similar to Kubernetes cluster locator for EKS) that would populate AWS accounts as resource entities in the catalog from something like AWS Organizations.  However, I don't think an AWS account catalog entity is useful for this particular plugin as a source of account credential information.  The catalog is not the right place to put account secrets like access key ID, secret access key, or role external ID.

Instead, my preferred approach is use the `accountDefaults` section to specify the default role name for any account, when it's not scalable to put every AWS account in the static config.

For example:
```yaml
# app-config.yaml
aws:
  accountDefaults:
    roleName: 'my-backstage-role'
    externalId: 'my-id'
```
In this example, I don't have to specify any account IDs in the static config, as long as all of my accounts have a role named `my-backstage-role` that can be assumed by my Backstage app.

Reply:
Yep, absolutely no secrets in the catalog! 😁 👍